pkg/serverless/trace/propagation/carriers.go

```diff
@@ -164,10 +165,22 @@ func snsSqsMessageCarrier(event events.SQSMessage) (tracer.TextMapReader, error)
 // snsEntityCarrier returns the tracer.TextMapReader used to extract trace
 // context from the attributes of an events.SNSEntity type.
 func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
-	msgAttrs, ok := event.MessageAttributes[datadogSQSHeader]
-	if !ok {
-		return nil, errorNoDDContextFound
+	// Check if this is a regular SNS message with Datadog trace information
+	if msgAttrs, ok := event.MessageAttributes[datadogTraceHeader]; ok {
+		return handleRegularSNSMessage(msgAttrs)
```

Comment:
Instead of naming these `handleSomething`, I'd like to keep them consistent with the rest of this file. The form is `<object-type>Carrier`, so this would be something like `snsMessageCarrier`.

Reply:
Alternatively, you could switch the flow around a bit, giving a slightly smaller diff.

```go
func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapREader, error) {
  if err := json.Unmarshall([]byte(event.Message), &eventBridgeEvent); err == nil {
    return eventBridgeCarrier(eventBridgeEvent)
  }

  ... keep everything else the same
```

That would change the order of precedence though, so please confirm this would work how you'd expect.

Reply:
I agree, this is a lot better and gives a smaller diff. It also significantly reduces the diff in `carriers_test.go`. 

From manual testing, it looks like unmarshalling to EventBridge always 'works' without error even if it was a regular SNS message, but the `eventBridgeEvent` is completely empty. So I had to add another check:
```go
func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
	// Check if this is an EventBridge event sent through SNS
	var eventBridgeEvent events.EventBridgeEvent
	if err := json.Unmarshal([]byte(event.Message), &eventBridgeEvent); err == nil {
		if len(eventBridgeEvent.Detail.TraceContext) > 0 {
			return eventBridgeCarrier(eventBridgeEvent)
		}
	}

	// If not, check if this is a regular SNS message with Datadog trace information
        ... everything else the same
```